filebeat/check.py

```diff
@@ -27,15 +27,23 @@ def check(self, instance):
 
         registry_contents = self._parse_registry_file(registry_file_path)
 
-        for item in registry_contents.itervalues():
+        for item in registry_contents:
+            self.log.debug("found item: %s" % item)
             self._process_registry_item(item)
 
     def _parse_registry_file(self, registry_file_path):
         try:
-            with open(registry_file_path) as registry_file:
-                return simplejson.load(registry_file)
-        except IOError:
-            self.log.warn('No filebeat registry file found at %s' % (registry_file_path, ))
+            # check that the collector user has permissions to read the file
+            if os.access(registry_file_path ,os.R_OK):
+                with open(registry_file_path) as registry_file:
+                    return ast.literal_eval(registry_file.read()) # convert the content from str to list
+            else:
+                # no permissions, alert and return empty dict
+                self.log.error("Cannot open the registry log file - check the file permissions")
+                return {}
+        except Exception as e:
```

Comment:
Bad form to catch any exception. Here the only expected type of exception is IOError, no point in catching more than that.

Reply:
And as to the permission thing, would be better to handle that here:
```
except IOError as e:
  if e.errno == errno.EACCES:
    self.log.error('Cannot open the registry log file - check the file permissions')
  else:
    ...
```

And on second thought since all you care about here is logging, there's no reason to do any kind of special handling for permission issues, just doing
```
except IOError as e:
  self.log.error('Cannot read the registry log file at %s: %s' % (registry_file_path, e))
```
will be plenty explicit enough:
```
Cannot read the registry log file at /var/lib/filebeat/registry: [Errno 13] Permission denied: '/var/lib/filebeat/registry'
```